src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1915,14 +1927,14 @@ void MergeTreeData::clearPartsFromFilesystemImpl(const DataPartsVector & parts_t
         ThreadPool pool(num_threads);
 
         /// NOTE: Under heavy system load you may get "Cannot schedule a task" from ThreadPool.
+        LOG_DEBUG(log, "Removing {} parts from filesystem: {} (concurrently)", parts_to_remove.size(), fmt::join(parts_to_remove, ", "));
```

Comment:
There was some concerns recently about too long messages and the time the mutex is taken (#41813), maybe log here with debug level, and each part with a trace level?

Reply:
Yes, writing one message per part will block the log mutex less time but it will block it many times (as many times as parts) which overall is much worse.

* Before:
  - Nº of messages: nº of parts
  - Size per message: (134 chars + part-name) * 
* After:
  - Nº of messages: 1
  - Size per message: 140 + (part->name * nº of parts)

So for 20 parts for example, before it'd block the mutex 20 times and each one it'd print around 150 chars. Now it will lock the mutex only once and flush ~440 characters. At least in **this case** the new behaviour seems way better.

In the PR that you link (https://github.com/ClickHouse/ClickHouse/pull/41813) the problem appears to be that the log keeps growing, so it keeps taking more and more time. That PR isn't even solving anything; it's just yielding the mutex more often so it doesn't block other threads for too long.

If avoiding long messages is necessary and desirable then it would make sense to add the limitation in `LOG_IMPL` or  `OwnSplitChannel` (and either truncate or split them). Otherwise the responsibility falls to the callers (which are way too many). What do you think?

Reply:
Actually I don't have any strong feelings here, but to me long log messages looks icky, but I'm fine with this change.

>That PR isn't even solving anything; it's just yielding the mutex more often so it doesn't block other threads for too long.

It writes only first X files, and replace everything other with a `...`, so it kind of does.

>If avoiding long messages is necessary and desirable then it would make sense to add the limitation in LOG_IMPL or OwnSplitChannel (and either truncate or split them). Otherwise the responsibility falls to the callers (which are way too many). What do you think?

Actually I'm not sure from which this limit raised from, though long log messages indeed does not makes a lot of sense to me. So adding a check into `LOG_IMPL` in this case sounds logical, though not sure that it worth it.

Reply:
Let's go ahead with this for now. If we feel the messages became too long I'll work to either batch them (prior to the call), truncate them (in `LOG_IMPL`?) or some other alternative.